src/support.sch

```diff
@@ -116,6 +116,19 @@
         </rule>
     </pattern>
 
+    <pattern id="day">
+        <rule context="day[not(number(.)=.)]">
+            <assert test="true" role="warn">
+                &lt;<name/>&gt;<value-of select="."/>&lt;/<name/>&gt; is ignored.
+            </assert>
+        </rule>
+        <rule context="day[number(.)=.]">
+            <assert test="count(preceding-sibling::day[number(.)=.])=0" role="warn">
+                Extra &lt;<name/>&gt; is ignored.
+            </assert>
+        </rule>
+    </pattern>
+
```

Comment:
This will work, but it seems like a bit of strange way of writing the tests. Is there are reason you haven't used one `rule` element containing both tests?

Something like this:
```
<pattern id="day">
<rule context="day">
            <assert test="number(.)=." role="warn">
                &lt;<name/>&gt;<value-of select="."/>&lt;/<name/>&gt; is ignored.
            </assert>
            <assert test="count(preceding-sibling::day)=0" role="warn">
                Extra &lt;<name/>&gt; is ignored.
            </assert>
        </rule>
</pattern>
```
(Same question for month and year)

Reply:
That changes the result a bit: currently we ignore any non-numerical elements (which might come first). Like https://github.com/libero/jats-support/blob/df06fa82452b3900d96a1ad26cfb2e3484427d91/tests/cases/pub-date/contents.xml#L4-L21

Reply:
I see, changing the second assert to something like the below would also work, but I'll leave it to you to decide which one makes more sense.

```
<assert test="(number(.)=.) and count(preceding-sibling::day[number(.)=.])=0" role="warn">
                Extra &lt;<name/>&gt; is ignored.
            </assert>
```

Reply:
Manage to simplify a bit further in c906ee5d83057d54eccb469ae61416b8ee50fe6f. Thanks!